sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -192,16 +216,63 @@ export class IdentityClient extends ServiceClient implements INetworkModule {
     }
   }
 
+  // Here is a custom layer that allows us to abort requests that go through MSAL,
+  // since MSAL doesn't allow us to pass options all the way through.
+
+  generateAbortSignal(correlationId?: string): AbortSignalLike {
+    const controller = new AbortController();
+    const key = correlationId || noCorrelationId;
+
+    const controllers = this.abortControllers.get(key) || [];
+    controllers.push(controller);
+    this.abortControllers.set(key, controllers);
+
+    return controller.signal;
+  }
+
+  abortRequests(correlationId: string = noCorrelationId): void {
+    const key = correlationId || noCorrelationId;
+    const controllers = [
+      ...(this.abortControllers.get(key) || []),
+      // MSAL passes no correlation ID to the get requests...
+      ...(this.abortControllers.get(noCorrelationId) || [])
+    ];
+    if (!controllers.length) {
+      return;
+    }
+    for (const controller of controllers) {
+      controller.abort();
+    }
+    this.abortControllers.set(key, undefined);
+    this.abortControllers.set(noCorrelationId, undefined);
+  }
+
+  getCorrelationId(options?: NetworkRequestOptions): string | undefined {
```

Comment:
how is this `string | undefined` when it's supposed to default to `noCorrelationId` ?

Reply:
MSAL handles the default case for us on v2, I copied this file over

Reply:
I’ll clean this up!

Reply:
I’ll remove this correlationId code. It comes from v2. I’ll address this in v2: https://github.com/Azure/azure-sdk-for-js/issues/16436